src/displayapp/screens/WatchFacePrideFlag.cpp

```diff
@@ -238,19 +249,16 @@ bool WatchFacePrideFlag::OnButtonPushed() {
 void WatchFacePrideFlag::UpdateScreen(Pinetime::Controllers::Settings::PrideFlag prideFlag) {
   themeChanged = true;
   auto prideFlagAsInt = static_cast<uint8_t>(prideFlag);
-  if (initialized) {
-    for (int i = 0; i < numBackgrounds; i++) {
-      lv_obj_del(backgroundSections[i]);
-    }
-    delete[] backgroundSections;
+  for (int i = 0; i < backgroundSections.size(); i++) {
+    lv_obj_del(backgroundSections[i]);
```

Comment:
with the use of `std::vector` we can now get rid of the indexed access and switch to a for-each loop (please check my syntax, didn't compile it :sweat_smile: )


```suggestion
  for (auto backgroundSection: backgroundSections) {
    lv_obj_del(backgroundSections[i]);
```